src_c/draw.c

```
@@ -972,9 +972,15 @@ rect(PyObject *self, PyObject *args, PyObject *kwargs)
             rect->y += rect->h;
             rect->h = -rect->h;
         }
+        // Ignore negative values
+        radius = MAX(radius, 0);
```

Instead of simply ignoring negative values here, we could raise an error if this is negative instead

---

That could be another solution indeed

---

silently clamping it to 0 vs raising an error

I'm not sure which is preferable when using the code. I tend to be against silent things. Make is explicit, not implicit.